src/app/components/tree-view/tree-node.component.ts

```diff
@@ -124,17 +127,31 @@ export class TreeNodeComponent extends ComponentBase implements OnInit {
         });
 
         this.tree.addNode(this);
+
+        this.tree.getExpandedNodes(this.key).subscribe(values => {
+            const found = values.find(value => this.node.id === value);
+            if (found) {
+                // Parameter is false in order not to persist state whether node is expanded or collapsed
+                this.toggleExpansion(false);
+            }
+        });
     }
 
-    public toggleExpansion() {
+    public toggleExpansion(toPersist: boolean) {
```

Comment:
Maybe rename to `shouldPersist` or just `persist`?
Since the default behaviour relies on it being falsy, you can explicitly declare it as `toggleExpansion(persist = false) { `